### PR TITLE
Tests for add, bracket notation and ip

### PR DESCRIPTION
**Pull request by Zonico6**
Tests 'add' as well as parsing and printing bracket notation. Tests ip parsing and printing. Doesn't include cases testing for malformed input or unsuccessfull addition.

### CODE REVIEW

**Comment by EinsimplerGeist** [comment]
This is amazing. So many probe strings. Please merge this!

File: src/test/java/edu/kit/informatik/IPTestMNT.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            } catch (ParseException e) {
+                e.printStackTrace();
+                fail();
+                return null;
+            }
+        }).toArray(IP[]::new);
+        for (int i = 0; i < ips.length; i++) {
+            for (int j = 0; j < ips.length; j++) {
+                int res = ips[i].compareTo(ips[j]);
+                assertEquals(res, Integer.compare(i, j));
```
Comment (lscobe16) [inline src/test/java/edu/kit/informatik/IPTestMNT.java:51]: I think, this test is too specific. Only the sign has to be the same. [Reference](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Comparable.html)

Comment (EinsimplerGeist) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: its a quite specific edgecase, yes. But its good to have a test that probes this, even if you can ignore it if you have worse problems.

Comment (lscobe16) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: I mean line 51 specifically.
According to contract, only the sign of the return value of compareTo is relevant. For example, you are allowed to always return 42 if the first IP is greater than the second, but this test will fail, although it is a valid (albeit weird) implementation.

Comment (lscobe16) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: You could wrap both values in Math.signum to fix it

Comment (EinsimplerGeist) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: thats true sorry, used to gitlab where you can mark whole paragraphs. Seemed like you mean the whole codepart

Comment (lscobe16) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: Yeah, I too think it's weird it also shows the previous lines. I only marked line 51

Comment (EinsimplerGeist) [reply src/test/java/edu/kit/informatik/IPTestMNT.java:51]: Github being github

**Pull request merged**